cmd/sealer/cmd/cluster/run.go

```diff
@@ -240,6 +240,15 @@ func createNewCluster(imageEngine imageengine.Interface, cf clusterfile.Interfac
 
 	logrus.Infof("start to create new cluster with image: %s", clusterImageName)
 
+	cmds := infraDriver.GetClusterLaunchCmds()
+	appNames := infraDriver.GetClusterLaunchApps()
+
+	// merge to application between v2.ClusterSpec, v2.Application and image extension
+	v2App, err := application.NewV2Application(v2.ConstructApplication(cf.GetApplication(), cmds, appNames), imageSpec.ImageExtension)
```

Comment:
lazy init the v2app

Reply:
i thought , if init app error ,return fast.